spec/models/generic_announcement_spec.rb

```diff
@@ -62,5 +62,22 @@
         expect(announcements).to include(*active_instance_announcements)
       end
     end
+
+    describe 'unread state' do
+      let(:creator) { create(:creator) }
+      let!(:user) { create(:user) }
+      let!(:system_announcement) { create(:system_announcement, creator: creator) }
```

Comment:
hmm... This tests if it marks the announcement as read for the creator who create this announcement. @lowjoel